dipy/align/_public.py

```diff
@@ -450,6 +452,19 @@ def affine_registration(moving, static,
                                 sigmas=sigmas,
                                 factors=factors)
 
+    # Convert pipeline to sanitized list of str
+    pipeline = list(pipeline)
+    for fi, func in enumerate(pipeline):
+        if not isinstance(func, str):
+            if callable(func):
```

Comment:
I am not sure that the first `if` is necessary. Why you do not keep only `if callable(func):`